src/index.js

```diff
@@ -207,10 +208,26 @@ function sanitizeHtml(html, options, _recursing) {
               }
             }
             if (name === 'iframe' && a === 'src') {
-              var whitelistedUrls = options.allowedIframeDomains.filter(function(url) {
-                return value.includes(url);
-              })
-              if (!whitelistedUrls.length) {
+              try {
+                var parsed = url.parse(value)
+                var whitelistedDomains = []
+                each(parsed, function(val, prop) {
+                  if (prop === 'hostname') {
+                    whitelistedDomains = options.allowedIframeDomains.filter(function(domain) {
+                      return val.includes(domain)
+                    })
+                  }
+                })
+        
+                if (!whitelistedDomains.length) {
+                  delete frame.attribs[a];
+                  return;
+                } else {
+                  value = url.format(parsed)
```

Comment:
No need to rebuild the URL with `format` since you never modified `parsed`. You can use the original value.

Reply:
Removed